docs: improve docs of elements using DataItem interface (#948)
Refinitiv/refinitiv-ui: documents/src/pages/elements/combo-box.md

`ef-combo-box` displays a text input and an associated pop-up element that helps users set a value.

## Usage

> The `ef-combo-box` uses the `data` property of [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) interface that extends from [Data Item](./custom-components/utils/data-management#data-item) interface.

Reviewer: ```suggestion
The `ef-combo-box` uses the `data` property `ComboBoxData` interface that extends from [Data Item](./custom-components/utils/data-management#data-item) interface.
```

`ef-combo-box` displays a text input and an associated pop-up element that helps users set a value.

## Usage

> The `data` property of the `ef-combo-box` use the [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) interface for its data items. Each item defaults to type ItemData, it can be anything extended from [DataItem](./custom-components/utils/data-management#data-item).

Reviewer 2: ```suggestion
The `data` property of the `ef-combo-box` use the [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) interface for its data items. Each item defaults to type ItemData, otherwise it can be any interface that extended from [DataItem](./custom-components/utils/data-management#data-item).
```

> The `data` property of the `ef-combo-box` use the [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) type for its data items. Each of this item defaults to type `ItemData`. It could also be anything extended from [DataItem](./custom-components/utils/data-management#data-item).

Reviewer 3: ```suggestion
The `data` property of the `ef-combo-box` uses the [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) type for its data items. Each of this item defaults to type `ItemData`. It could also be anything extended from [DataItem](./custom-components/utils/data-management#data-item).
```

> The `data` property of the `ef-combo-box` use the [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) interface for its data items. Each item defaults to type ItemData, otherwise it can be any interface that extended from [DataItem](./custom-components/utils/data-management#data-item).

Reviewer 3: Apply this pattern to all elements covered by this update.

```suggestion
The `data` property of the `ef-combo-box` use the [ComboBoxData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/combo-box/helpers/types.ts) type for its data items. Each of this item defaults to type ItemData. It could also be anything extended from [DataItem](./custom-components/utils/data-management#data-item).
```

Reviewer 2: +1 to this kub
